packages/core/src/sdk/product/usePageProductsQuery.ts

```diff
@@ -64,14 +67,27 @@ export const query = gql(`
 
 const getKey = (object: any) => JSON.stringify(object)
 
+interface UseCreateUseGalleryPageProps {
+  initialPages: ClientManyProductsQueryQuery
+  serverManyProductsVariables: ClientManyProductsQueryQueryVariables
+}
+
 /**
  * Use this hook for managed pages state and creating useGalleryPage hook that will be used for fetching a list of products per pages in PLP or Search
  */
-export const useCreateUseGalleryPage = () => {
-  const [pages, setPages] = useState<ClientManyProductsQueryQuery[]>([])
+export const useCreateUseGalleryPage = (
+  params?: UseCreateUseGalleryPageProps
+) => {
+  const initialPages = params?.initialPages?.search ? [params.initialPages] : []
+  const initialVariables = params?.serverManyProductsVariables
+    ? [getKey(params.serverManyProductsVariables)]
+    : []
+
+  const [pages, setPages] =
+    useState<ClientManyProductsQueryQuery[]>(initialPages)
   // We create pagesRef as a mirror of the pages state so we don't have to add pages as a dependency of the useGalleryPage hook
-  const pagesRef = useRef<ClientManyProductsQueryQuery[]>([])
-  const pagesCache = useRef<string[]>([])
+  const pagesRef = useRef<ClientManyProductsQueryQuery[]>(initialPages)
+  const pagesCache = useRef<string[]>(initialVariables)
```

Comment:
we start the pages and the cache from PLPs with the values from server in page 0